### PR TITLE
Make batch size as a option

### PR DESCRIPTION
**Pull request by hoaleee**
Make batch size as a option.

## Motivation

We are developing a network based on zkevm - a new evm platform. We are trying to increase compatibility of MUD Framework and zkevm. We found that with the default batch size of 100, zkevm may not be able to handle it, with the --slow option the waiting time is too long. 

## Solution

So we added batch size as an option to increase customization compatibility with many chains.

### CODE REVIEW

File: crates/script/src/broadcast.rs [hunk 1]
```diff
@@ -321,7 +321,8 @@ impl BundledState {
 
                 // We send transactions and wait for receipts in batches of 100, since some networks
                 // cannot handle more than that.
-                let batch_size = if sequential_broadcast { 1 } else { 100 };
+                let valid_batch_size = if self.args.batch_size > 100 { 100 } else { self.args.batch_size };
```
Comment (DaniPopes) [inline crates/script/src/broadcast.rs:324]:
```suggestion
                let valid_batch_size = self.args.batch_size.min(100);
```

File: crates/script/src/lib.rs [hunk 2]
```diff
@@ -105,8 +105,14 @@ pub struct ScriptArgs {
 
     /// Broadcasts the transactions.
     #[arg(long)]
+
```
Comment (DaniPopes) [inline crates/script/src/lib.rs:108]:
```suggestion
```

File: crates/script/src/lib.rs [hunk 3]
```diff
@@ -105,8 +105,14 @@ pub struct ScriptArgs {
...
     pub broadcast: bool,
 
+    /// Batch size of transactions.
+    /// --batch-size
```
Comment (DaniPopes) [inline crates/script/src/lib.rs:112]:
```suggestion
```

File: crates/script/src/lib.rs [hunk 4]
```diff
@@ -105,8 +105,14 @@ pub struct ScriptArgs {
...
+    #[arg(long, default_value = "100")]
+    pub batch_size: i32,
```
Comment (DaniPopes) [inline crates/script/src/lib.rs:114]:
```suggestion
    pub batch_size: usize,
```

File: crates/script/src/broadcast.rs [hunk 5]
```diff
@@ -321,7 +321,8 @@ impl BundledState {
...
+                let batch_size = if sequential_broadcast { 1 } else { valid_batch_size as usize };
```
Comment (DaniPopes) [inline crates/script/src/broadcast.rs:325]:
```suggestion
                let batch_size = if sequential_broadcast { 1 } else { valid_batch_size };
```

**Review by DaniPopes — CHANGES_REQUESTED** [review]
cc @klkvr

**Review by klkvr — APPROVED** [review]

**Review by DaniPopes — APPROVED** [review]
Thanks!

**Pull request merged**